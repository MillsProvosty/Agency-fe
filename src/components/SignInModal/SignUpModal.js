import React, { Component } from "react";
import "./SignUpModal.css";
import { FaHandsHelping } from "react-icons/fa";

export class SignUpModal extends Component {
  constructor() {
    super();
    this.state = {};
  }

  render() {
    return (
      <form>
        <p>Thanks for Signing Up!</p>
        <input
          type="text"
          placeholder="Enter your name"
          name="name"
          value={this.state.name}
        />
        <input
          type="text"
          placeholder="Enter your email"
          name="email"
          value={this.state.email}
        />
        <input
          type="text"
          placeholder="Enter your phone"
          name="phone"
          value={this.state.phone}
        />
        <input
          type="text"
          placeholder="Enter your password"
          name="password"
          value={this.state.password}
        />
        <input
          type="text"
          placeholder="Confirm your password"
          name="confirmation"
          value={this.state.confirmation}
        />
        <button>Submit!</button>
      </form>
    );
    //pronouns
    //language
  }
}
