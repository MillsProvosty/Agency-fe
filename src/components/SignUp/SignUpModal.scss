.SignUpModal {
  margin: 50px auto;
  height: auto;
  background-color: white;
  border: 1px solid lightgrey;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 500px;
  padding: 20px;
  font-family: "Quicksand", sans-serif;
  color: darkblue;

  & p {
      font-size: 2em;
  }

  & input, button {
    border-radius: 5px;
    font-size: 1.5em;
    height: 2em;
    border: 1px solid darkgrey;
    padding: 5px;
    width: 300px;
    font-family: 'Quicksand', sans-serif;
    margin-top: 10px;
}

& button {
    border: 2px solid white;
    color: white;
    background-color: darkblue;
    margin-top: 30px
  }

  & button:hover {
    border: 2px solid darkblue;
    color: darkblue;
    background-color: white;
  }
}

@media screen and (max-width: 375px) {
  .SignUpModal {
    margin: 130px 4px;
    height: auto;
    width: 98%;
    padding: 10px;
  }
}

@media screen and (display-mode: standalone) {
  .SignUpModal {
    margin: 130px 4px;
    height: auto;
    width: 98%;
    padding: 10px;
  }
}