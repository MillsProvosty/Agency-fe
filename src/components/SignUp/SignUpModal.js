import React, { useState, useEffect } from "react";
import "./SignUpModal.scss";
import { Link } from "react-router-dom";
import styled from 'styled-components';
import {validate} from '../../hooks/signInFormValidationRules';
import { useSignInForm } from '../../hooks/useForm';
import { postAUser } from '../../util/apiCalls';
import { setUser } from '../../actions';
import { connect } from "react-redux";

const SignUpForm = styled.form`
  margin: 50px auto;
  height: auto;
  background-color: white;
  border: 1px solid lightgrey;
  border-radius: 8px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  width: 500px;
  padding: 20px;
  font-family: "Quicksand", sans-serif;
  color: darkblue;
  
  @media screen and (max-width: 375px) {
    margin: 130px 4px;
    height: auto;
    width: 98%;
    padding: 10px;
}

@media screen and (display-mode: standalone) {
    margin: 130px 4px;
    height: auto;
    width: 98%;
    padding: 10px;
}
`

const PTag = styled.p`
  font-size: 2em;
`

const Input = styled.input`
  border-radius: 5px;
  font-size: 1.5em;
  height: 2em;
  border: 1px solid darkgrey;
  padding: 5px;
  width: 300px;
  font-family: 'Quicksand', sans-serif;
  margin-top: 10px;
`

const Button = styled.button`
  border-radius: 5px;
  font-size: 1.5em;
  height: 2em;
  padding: 5px;
  width: 300px;
  font-family: 'Quicksand', sans-serif;
  margin-top: 10px;
  border: 2px solid white;
  color: white;
  background-color: darkblue;
  margin-top: 30px

  :hover {
  border: 2px solid darkblue;
  color: darkblue;
  background-color: white;
  }
`

export const SignUpModal = (props) => {
  const [ disabled, setDisabled ] = useState(true)
  const { values, handleChange } = useSignInForm(validate);

  const setUser = async (values) => {
    try {
      const user = await postAUser(values)
      props.setAUser(user)
    } catch(error) {
      console.log(error)
    }
  }
  
  function setSetDisabled() {
    if (!values.error) {
      setDisabled(false)
    } else {
      setDisabled(true);
    }
  }

  useEffect(() => {
      validate(values)
      if(!values.error){
        setSetDisabled()
      }
    }, [values]);

  return (
    <SignUpForm className="SignUpModal">
      <PTag>Thanks for Signing Up!</PTag>
      <Input
        type="text"
        placeholder="Enter your First name"
        name="firstname"
        value={values.firstname || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
       <Input
        type="text"
        placeholder="Enter your Last name"
        name="lastname"
        value={values.lastname || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <Input
        type="text"
        placeholder="Enter your email"
        name="email"
        value={values.email || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <Input
        type="text"
        placeholder="Enter your phone"
        name="phone"
        value={values.phone || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <Input
        type="text"
        placeholder="Enter your password"
        name="password"
        value={values.password || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      <Input
        type="text"
        placeholder="Confirm your password"
        name="confirmation"
        value={values.confirmation || ""}
        onChange={handleChange}
        autoComplete="off"
        required
      />
      {/* {errors && <p>{errors}</p>} */}
      <Link to="/profile">
        <Button disabled={disabled} onClick={() => setUser(values)}>Submit!</Button>
      </Link>
    </SignUpForm>
  );
};

export const mapDispatchToProps = dispatch => ({
  setAUser: user => dispatch(setUser(user))
})

export default connect(null, mapDispatchToProps)(SignUpModal)
