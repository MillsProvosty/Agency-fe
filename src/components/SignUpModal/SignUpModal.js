import React, { Component } from "react";
import "./SignUpModal.css";
import { NavLink } from 'react-router-dom';

export class SignUpModal extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <form>
        <p>Thanks for Signing Up!</p>
        <input
          type="text"
          placeholder="Enter your name"
          name="name"
          value={this.state.name}
        />
        <input
          type="text"
          placeholder="Enter your email"
          name="email"
          value={this.state.email}
        />
        <input
          type="text"
          placeholder="Enter your phone"
          name="phone"
          value={this.state.phone}
        />
        <input
          type="text"
          placeholder="Enter your password"
          name="password"
          value={this.state.password}
        />
        <input
          type="text"
          placeholder="Confirm your password"
          name="confirmation"
          value={this.state.confirmation}
        />
        <NavLink to='/profile'><button>Submit!</button></NavLink>
      </form>
    );
    //pronouns
    //language
  }
}
