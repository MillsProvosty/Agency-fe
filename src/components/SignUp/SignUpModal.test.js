import React from 'react';
import { shallow } from 'enzyme';
import { SignUpModal } from './SignUpModal';

describe('SignUpModal', () => {
  let wrapper; 
  
  const mockSetUser = jest.fn()
  const mocksetAllOpportunities = jest.fn()

  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState')
  useStateSpy.mockImplementation((init) => [init, setState]);

  beforeEach(() => {
    wrapper = shallow(
      <SignUpModal 
        role='client'
        setAUser={mockSetUser}
        setAllOpps={mocksetAllOpportunities}
      />
    )
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  }) 

  // it('should update state')
})
