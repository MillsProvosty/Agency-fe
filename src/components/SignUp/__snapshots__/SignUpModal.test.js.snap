// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpModal should match snapshot 1`] = `
<styled.form
  className="SignUpModal"
>
  <styled.p>
    Thanks for Signing Up!
  </styled.p>
  <styled.input
    autoComplete="off"
    name="firstname"
    onChange={[Function]}
    placeholder="Enter your First name"
    required={true}
    type="text"
    value=""
  />
  <styled.input
    autoComplete="off"
    name="lastname"
    onChange={[Function]}
    placeholder="Enter your Last name"
    required={true}
    type="text"
    value=""
  />
  <styled.input
    autoComplete="off"
    name="email"
    onChange={[Function]}
    placeholder="Enter your email"
    required={true}
    type="text"
    value=""
  />
  <styled.input
    autoComplete="off"
    name="phone"
    onChange={[Function]}
    placeholder="Enter your phone"
    required={true}
    type="text"
    value=""
  />
  <styled.input
    autoComplete="off"
    name="password"
    onChange={[Function]}
    placeholder="Enter your password"
    required={true}
    type="password"
    value=""
  />
  <styled.input
    autoComplete="off"
    name="confirmation"
    onChange={[Function]}
    placeholder="Confirm your password"
    required={true}
    type="password"
    value=""
  />
  <styled.button
    disabled={true}
    onClick={[Function]}
  >
    Submit!
  </styled.button>
</styled.form>
`;
